pkg/controller/controller_binding.go

```diff
@@ -347,38 +350,69 @@ func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServiceP
 
 func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[string]interface{}) error {
 	glog.V(5).Infof("Creating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      binding.Spec.SecretName,
-			Namespace: binding.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
 
+	secretData := make(map[string][]byte)
 	for k, v := range credentials {
 		var err error
-		secret.Data[k], err = serialize(v)
+		secretData[k], err = serialize(v)
 		if err != nil {
 			return fmt.Errorf("Unable to serialize credential value %q: %v; %s",
 				k, v, err)
 		}
 	}
 
-	found := false
+	secretClient := c.kubeClient.Core().Secrets(binding.Namespace)
+	// Retry loop for upserting the Secret
```

Comment:
The whole CREATE/UPDATE logic for Secret is rewritten:
- retry loop
- explicitly checking the errors (`IsConflict`, `IsAlreadyExists`)
- updating existing Secret instead of completely overwriting it
  - this automatically makes it safe for conflicting updates because of `resourceVersion` check on server side
  - we don't overwrite any metadata which can be potentially changed by other controllers (annotations, labels, ownerReferences etc)